### DISCUSSION TITLE
v-infinite-scroll onLoad function parameter type error

### DISCUSSION THREAD

**Discussion by q962** [vuetifyjs/vuetify#20503] [Q&A]
```
// ts(7031)
// (parameter) done: any
async function loadnext({ done }) {}
```

How to eliminate this error?

**Comment by frederikheld** [comment]
Can you please give more context to your code snippet? The onload event of `v-infinite-scroll` is `@load` ([see docs](https://vuetifyjs.com/en/api/v-infinite-scroll/#events-load)). I assume that you use your function like this:

```
<v-infinite-scroll @load="loadnext" />
```

So the typing should be

```
async function loadnext({ done: (status: 'loading' | 'error' | 'empty' | 'ok') => void }) {}
```

**Comment by q962** [comment]
After learning, ts requires that the function parameter types must be clear.

**Reply by frederikheld** [reply]
Depends on your level of type checking. But its good practice to not just implicitly type stuff as `any` because this defies the purpose of TypeScript.

But usually TS can infer types in situations like these which means that it figures out the type by looking where a value comes from along the call stack. So in many situations you do not need to type variables or parameters as TypeScript already knows them. But it's still good practice to do so in situations where this is not obvious to the developer.

See: https://www.typescriptlang.org/docs/handbook/type-inference.html
